### ISSUE TITLE
Auto batch size

### ISSUE THREAD

**Issue by wolfie82** [Deci-AI/super-gradients#1770]
### 💡 Your Question

Hello, I see in the documentation that [Pre launch callbacks](https://docs.deci.ai/super-gradients/latest/docstring/training/pre_launch_callbacks.html) have something called `AutoTrainBatchSizeSelectionCallback` to auto-tune the hyper parameters.  It looks like the mode of execution has to be via yaml configuration.  Is there a way to configure this by setting a config within the training_params as I'm running this in Co-Lab for now?

### Versions

Hypervisor vendor:                  KVM
Virtualization type:                full
L1d cache:                          192 KiB (6 instances)
L1i cache:                          192 KiB (6 instances)
L2 cache:                           6 MiB (6 instances)
L3 cache:                           38.5 MiB (1 instance)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-11
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Vulnerable; SMT Host state unknown
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Vulnerable
Vulnerability Retbleed:             Vulnerable
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Vulnerable
Vulnerability Spectre v1:           Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:           Vulnerable, IBPB: disabled, STIBP: disabled, PBRSB-eIBRS: Vulnerable
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Vulnerable

Versions of relevant libraries:
[pip3] numpy==1.23.0
[pip3] onnx==1.13.0
[pip3] onnx-simplifier==0.4.35
[pip3] onnxruntime==1.13.1
[pip3] pytorch-quantization==2.1.2
[pip3] torch==2.1.0+cu121
[pip3] torchaudio==2.1.0+cu121
[pip3] torchdata==0.7.0
[pip3] torchmetrics==0.8.0
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.16.0
[pip3] torchvision==0.16.0+cu121
[pip3] triton==2.1.0
[conda] Could not collect

**Comment by shaydeci** [comment]
@wolfie82 pre_launch_callbacks modify the config when lanching training with yaml configuration files.
At the moment execution has to be via yaml configuration for it to be used.
You can, however define YAML configuraiton files in your Colab notebook and call train_from_config.
Other workarounds (like converting dicts to DictConfig and calling train_from_config) might be possible too.
